GetChromeVersions task: Try to find older versions, if the latest one is

radical opened the pull request:
.. not available.

Sometimes the version obtained can have an inexplicable value for
branch_position which causes the task to fail to find a corresponding
snapshot.

Instead, use `https://omahaproxy.appspot.com/history` to get the last
few versions, and try the previous ones if the latest one cannot be
found.

This returns a csv like:
```csv
win,stable,113.0.5672.93,2023-05-08 20:43:02.119774
win64,stable,113.0.5672.93,2023-05-08 20:43:01.624636
mac_arm64,stable,113.0.5672.92,2023-05-08 20:43:01.067910
```

.. and the version here can be used with
`https://omahaproxy.appspot.com/deps.json?version=` to get the branch
position, and other details.

And limit the search to latest 2 major versions.

kg commented on src/tasks/WasmBuildTasks/GetChromeVersions.cs:
```diff
@@ -78,38 +82,85 @@ private async Task<bool> ExecuteInternalAsync()
 
-        int branchPosition = int.Parse(version.branch_base_position);
-        for (int i = 0; i < s_numBranchPositionsToTry; i++)
+        await foreach (string version in GetVersionsAsync())
```
Should probably set a limit on how many versions we try

radical replied:
Maybe it should also consider the major version, and only try like 2 previous major versions.

radical commented:
/azp run runtime-wasm

radical committed:
Rename ChromeUseLatestVersion to ChromeFindLatestAvailableVersion

radical committed:
GetChromeVersions task: Try to find older versions, if the latest one is

.. not available.

Sometimes the version obtained can have an inexplicable value for
branch_position which causes the task to fail to find a corresponding
snapshot.

Instead, use `https://omahaproxy.appspot.com/history` to get the last
few versions, and try the previous ones if the latest one cannot be
found.

This returns a csv like:
```csv
win,stable,113.0.5672.93,2023-05-08 20:43:02.119774
win64,stable,113.0.5672.93,2023-05-08 20:43:01.624636
mac_arm64,stable,113.0.5672.92,2023-05-08 20:43:01.067910
```

.. and the version here can be used with
`https://omahaproxy.appspot.com/deps.json?version=` to get the branch
position, and other details.

And limit the search to latest 2 major versions.

radical commented:
/azp run runtime-wasm

radical commented:
Failures are unrelated.
